feat: :sparkles: add top-nav component

chocoboy08 opened the pull request:
TopNav 컴포넌트 추가했습니다

## 개요
상단 네이게이션 컴포넌트
## 작업내용
로고는 아직 미정 /  네비게이션 바, 유저 프로필 기능 구현
## 주의사항
응애

strawji02 committed:
fix: :rotating_light: update lint rules

eslint no-unused-vars 설정을 off로 변경했습니다

strawji02 committed:
fix: :rotating_light: update lint rules

eslint no-unused-vars 설정을 off로 변경했습니다

strawji02 committed:
build: :wrench: update tsconfig noUnusedParam

tsconfig.json의 noUnusedParameters 속성을 false로 변경했습니다. 파라미터에서 props rest 인자를 항상 받기 위함입니다.

strawji02 committed:
build: :wrench: update tsconfig noUnusedParam

tsconfig.json의 noUnusedParameters 속성을 false로 변경했습니다. 파라미터에서 props rest 인자를 항상 받기 위함입니다.

strawji02 committed:
feat: :lipstick: add stack component

세로 정렬로 element들을 표시해주는 stack 컴포넌트를 추가했습니다

strawji02 committed:
feat: :lipstick: add stack component

세로 정렬로 element들을 표시해주는 stack 컴포넌트를 추가했습니다

chocoboy08 committed:
feat: :sparkles: add top-nav component

TopNav 컴포넌트 추가했습니다

chocoboy08 committed:
feat: :sparkles: add top-nav component

TopNav 컴포넌트 추가했습니다

strawji02 reviewed (commented):
일단 이렇게 많은데 커밋이 하나인거부터 어지러움

strawji02 committed:
build: :mute: add pre-commit hook that remove all js console.log

콘솔을 커밋 전에 없애주는 훅을 추가했습니다

strawji02 committed:
build: :mute: add pre-commit hook that remove all js console.log

콘솔을 커밋 전에 없애주는 훅을 추가했습니다

strawji02 reviewed (changes requested):
다시해오쇼.

chocoboy08 committed:
refactor: :recycle: refector TopNav component

TopNav 폴더 추가, User 파일명 변경, 코드 수정

chocoboy08 committed:
refactor: :recycle: refector TopNav component

TopNav 폴더 추가, User 파일명 변경, 코드 수정

strawji02 committed:
build: :hammer: update husky precommit

커밋 전  console.log 를 확인합니다

strawji02 committed:
build: :hammer: update husky precommit

커밋 전  console.log 를 확인합니다

strawji02 committed:
feat: :sparkles: add group component

children을 가로 정렬시켜주는 컴포넌트인 Group을 추가했습니다

strawji02 committed:
feat: :sparkles: add group component

children을 가로 정렬시켜주는 컴포넌트인 Group을 추가했습니다

strawji02 committed:
feat: :sparkles: banner making

...

strawji02 committed:
feat: :sparkles: banner making

...

strawji02 committed:
build: :wrench: update tsconfig.json paths

tsconfig.json path에 @, @components, @base 를 추가했습니다. vite 설정 파일을 추가했습니다.

strawji02 committed:
build: :wrench: update tsconfig.json paths

tsconfig.json path에 @, @components, @base 를 추가했습니다. vite 설정 파일을 추가했습니다.

strawji02 committed:
feat: :sparkles: add tabs component

Banner 컴포넌트를 삭제하고 공용 Tabs 컴포넌트를 제작합니다

strawji02 committed:
feat: :sparkles: add tabs component

Banner 컴포넌트를 삭제하고 공용 Tabs 컴포넌트를 제작합니다

chocoboy08 committed:
refactor: :recycle: 2th refactor TopNav component

TopNavBar, UserDropdownMenu children 사용, UserDropdownMenu css 수정

chocoboy08 committed:
refactor: :recycle: 2th refactor TopNav component

TopNavBar, UserDropdownMenu children 사용, UserDropdownMenu css 수정

strawji02 committed:
fix: :pencil2: fix vite config typo

vite.config.ts의 @components 의 오타를 수정했습니다

strawji02 committed:
fix: :pencil2: fix vite config typo

vite.config.ts의 @components 의 오타를 수정했습니다

strawji02 committed:
feat: :lipstick: add tabs component

Tabs base 컴포넌트를 추가했습니다

strawji02 committed:
feat: :lipstick: add tabs component

Tabs base 컴포넌트를 추가했습니다

strawji02 committed:
build: :wrench: update eslint useEffect rule

useEffect depth를 무시하는 eslint rule을 추가했습니다

strawji02 committed:
build: :wrench: update eslint useEffect rule

useEffect depth를 무시하는 eslint rule을 추가했습니다

strawji02 committed:
fix: :adhesive_bandage: update tabs css props bug

tabs 컴포넌트에서 css 사용시 렌더링되지 않는 문제를 수정했습니다

strawji02 committed:
fix: :adhesive_bandage: update tabs css props bug

tabs 컴포넌트에서 css 사용시 렌더링되지 않는 문제를 수정했습니다

strawji02 committed:
feat: :lipstick: update tabsButton component style props

Tabs 컴포넌트의 style 전용 prop를 추가했습니다

strawji02 committed:
feat: :lipstick: update tabsButton component style props

Tabs 컴포넌트의 style 전용 prop를 추가했습니다

strawji02 committed:
feat: add home component

Home 컴포넌트를 추가했습니다

strawji02 committed:
feat: add home component

Home 컴포넌트를 추가했습니다

strawji02 committed:
build: :wrench: update tsconfig paths

path에 @base를 추가했습니다

strawji02 committed:
build: :wrench: update tsconfig paths

path에 @base를 추가했습니다

strawji02 committed:
fix: :bug: update group, stack component props

css가 적용되지 않아 props에 className을 rest로 넘겨주게 수정했습니다

strawji02 committed:
fix: :bug: update group, stack component props

css가 적용되지 않아 props에 className을 rest로 넘겨주게 수정했습니다

strawji02 committed:
chore: :truck: update tabs component

Tabs 컴포넌트의 export를 index.ts에 추가하고, defaulValue의 값이 적용되지 않던 버그를 수정했습니다

strawji02 committed:
chore: :truck: update tabs component

Tabs 컴포넌트의 export를 index.ts에 추가하고, defaulValue의 값이 적용되지 않던 버그를 수정했습니다

strawji02 committed:
Merge pull request #4 from FRIED-NOTE/1-feature-add-home-component

add banner component

strawji02 committed:
build: :package: add file base routing

file base routing을 위해 react-router와 vite plugin을 추가했습니다

strawji02 committed:
build: :package: add file base routing

file base routing을 위해 react-router와 vite plugin을 추가했습니다

strawji02 committed:
feat: :sparkles: update route files

pages의 라우팅 파일을 업데이트했습니다

strawji02 committed:
feat: :sparkles: update route files

pages의 라우팅 파일을 업데이트했습니다

strawji02 committed:
docs: :memo: update README.md

README.md 파일 내용을 최신화했습니다

strawji02 committed:
docs: :memo: update README.md

README.md 파일 내용을 최신화했습니다

strawji02 committed:
chore: :fire: remove commnet

주석 삭제, page index에 예시 버튼 추가했습니다

strawji02 committed:
chore: :fire: remove commnet

주석 삭제, page index에 예시 버튼 추가했습니다

strawji02 committed:
Merge pull request #6 from FRIED-NOTE/5-feature-add-react-router

feature: add react router

chocoboy08 committed:
refactor: :recycle: 3rd refactoring TopNav component

디자인 수정, UserDropdownMenu 컴포넌트 삭제, TopNavBar와 TopNavUser 컴포넌트에 children과 props 사용

chocoboy08 committed:
refactor: :recycle: 3rd refactoring TopNav component

디자인 수정, UserDropdownMenu 컴포넌트 삭제, TopNavBar와 TopNavUser 컴포넌트에 children과 props 사용

chocoboy08 committed:
refactor: :recycle: 4th refectoring TopNav component

assets 폴더에 svg 파일 추가, NavBar 클릭 시 filledCircle 적용, 레이아웃 적용

chocoboy08 committed:
refactor: :recycle: 4th refectoring TopNav component

assets 폴더에 svg 파일 추가, NavBar 클릭 시 filledCircle 적용, 레이아웃 적용

strawji02 committed:
chore: :sparkles: add mergeRef util

ref 여러개를 인자로 받아 하나의 ref를 리턴해주는 mergeRef 함수를 추가했습니다

strawji02 committed:
chore: :sparkles: add mergeRef util

ref 여러개를 인자로 받아 하나의 ref를 리턴해주는 mergeRef 함수를 추가했습니다

strawji02 committed:
feat: :sparkles: add Popover component

팝오버 컴포넌트를 제작했습니다. pages/Home에서 예제를 확인하세요.

strawji02 committed:
feat: :sparkles: add Popover component

팝오버 컴포넌트를 제작했습니다. pages/Home에서 예제를 확인하세요.

strawji02 committed:
feat: :sparkles: add input box component

InputBox 컴포넌트를 추가했습니다

strawji02 committed:
feat: :sparkles: add input box component

InputBox 컴포넌트를 추가했습니다

strawji02 committed:
feat: :sparkles: add input box component

InputBox 컴포넌트를 추가했습니다

strawji02 committed:
chore: :coffin: update dead code

사용되지 않는 코드를 제거했습니다

strawji02 committed:
chore: :coffin: update dead code

사용되지 않는 코드를 제거했습니다

strawji02 committed:
refactor: :recycle: update clean code

rect의 복잡한 로직을 custom hooks로 별도 분리해 사용하고, context 파일을 분리했습니다.

strawji02 committed:
refactor: :recycle: update clean code

rect의 복잡한 로직을 custom hooks로 별도 분리해 사용하고, context 파일을 분리했습니다.

strawji02 committed:
Merge pull request #9 from FRIED-NOTE/8-feature-add-popover-component

feature: add popover component

strawji02 committed:
Merge branch '7-feature-add-search-bar' of https://github.com/My-Recipe/Frontend into 7-feature-add-search-bar

strawji02 committed:
build: :arrow_up: add hangul-js lib

한글js 라이브러리를 추가했습니다

strawji02 committed:
build: :arrow_up: add hangul-js lib

한글js 라이브러리를 추가했습니다

strawji02 committed:
feat: :sparkles: add inputbox component

inputBox 컴포넌트를 추가했습니다. Chip 제작 후 추가적인 작업이 필요합니다.

strawji02 committed:
feat: :sparkles: add inputbox component

inputBox 컴포넌트를 추가했습니다. Chip 제작 후 추가적인 작업이 필요합니다.

strawji02 committed:
build: :package: add storybook

storybook 패키지를 추가했습니다

strawji02 committed:
build: :package: add storybook

storybook 패키지를 추가했습니다

strawji02 committed:
feat: :lipstick: add typography design system

Design system의 typography, color를 적용했습니다

strawji02 committed:
feat: :lipstick: add typography design system

Design system의 typography, color를 적용했습니다

strawji02 committed:
feat: :lipstick: add button component

Button 컴포넌트를 추가했습니다

strawji02 committed:
feat: :lipstick: add button component

Button 컴포넌트를 추가했습니다

strawji02 committed:
chore: :bento: add icons assets

아이콘을 에셋에 추가했습니다

strawji02 committed:
chore: :bento: add icons assets

아이콘을 에셋에 추가했습니다

strawji02 committed:
chore: :technologist: update Group component props

Group 컴포넌트에 gap prop을 추가했습니다

strawji02 committed:
chore: :technologist: update Group component props

Group 컴포넌트에 gap prop을 추가했습니다

strawji02 committed:
feat: :sparkles: add Tag component

Tag 컴포넌트와 사용 예시를 추가했습니다

strawji02 committed:
feat: :sparkles: add Tag component

Tag 컴포넌트와 사용 예시를 추가했습니다

strawji02 committed:
chore: :pencil2: update Typography index export

base 컴포넌트에서 Typography export를 추가했습니다

strawji02 committed:
chore: :pencil2: update Typography index export

base 컴포넌트에서 Typography export를 추가했습니다

strawji02 committed:
feat: :sparkles: add Stroke component

Stroke 컴포넌트와 사용 예시를 추가했습니다

strawji02 committed:
feat: :sparkles: add Stroke component

Stroke 컴포넌트와 사용 예시를 추가했습니다

strawji02 committed:
chore: :pencil2: update icon filename

icon 에셋중 언더스코어로 잘못 써진 파일명을 수정했습니다

strawji02 committed:
chore: :pencil2: update icon filename

icon 에셋중 언더스코어로 잘못 써진 파일명을 수정했습니다

strawji02 committed:
Merge branch '10-feature-add-storybook' of https://github.com/My-Recipe/Frontend into 10-feature-add-storybook

strawji02 committed:
chore: :fire: remove storybook example code

chocoboy08 committed:
Merge pull request #12 from FRIED-NOTE/7-feature-add-search-bar

feature: add search bar

strawji02 committed:
refactor: :zap: update InputBox component value & onChange props

InputBox 컴포넌트에 value와 onChange props를 추가했습니다

strawji02 committed:
test: :hammer: add storybook stories

stories를 추가했습니다

chocoboy08 committed:
Merge pull request #15 from FRIED-NOTE/13-feature-add-components

feature: add components

strawji02 committed:
refactor: :recycle: update Tabs component CSS Props

CSS Props의 확장성을 위해 타입을 CSSProperties로 변경했습니다

더이상 ...Css 의 props에서 css() 함수를 사용하지 않습니다

strawji02 committed:
docs: :technologist: add tabs storybook

Tabs 컴포넌트의 스토리를 추가했습니다

strawji02 committed:
docs: :technologist: add Popover storybook

Popover 컴포넌트의 스토리를 추가했습니다

strawji02 committed:
docs: :technologist: update Group storybook

Group 컴포넌트의 스토리를 추가했습니다

strawji02 committed:
docs: :technologist: add Stack storybook

Stack 컴포넌트의 스토리를 추가했습니다

strawji02 committed:
docs: :technologist: add Stroke storybook

Stroke 컴포넌트의 스토리를 추가했습니다

strawji02 committed:
docs: :technologist: add typography storybook

Typography 컴포넌트의 스토리를 추가했습니다

strawji02 committed:
docs: :technologist: add Tag storybook

Tag 컴포넌트의 스토리를 추가했습니다

strawji02 committed:
docs: :memo: update storybook docs

strawji02 committed:
build: :construction_worker: add storybook chromatic build

strawji02 committed:
build: :green_heart: update chromatic.yml

strawji02 committed:
chore: :green_heart: update chromatic.yml

strawji02 committed:
refactor: :recycle: update Tag onClose prop

onClose 함수에서 mouseEvent도 넘겨주도록 변경했습니다

strawji02 committed:
refactor: :recycle: update Tabs component props

Tabs 컴포넌트의 buttonGroupProps를 optional로 변경했습니다

strawji02 committed:
docs: :memo: update storybook description

strawji02 committed:
build: :green_heart: update popover input test

strawji02 committed:
build: :green_heart: update chromatic.yml cache

strawji02 committed:
build: :green_heart: update chromatic.yml

캐싱 수정

strawji02 committed:
build: :green_heart: update chromatic.yml

strawji02 committed:
build: :green_heart: fix ci cache

strawji02 committed:
build: :green_heart: update chromatic build

strawji02 committed:
build: :green_heart: update chromatic.yml

strawji02 committed:
build: :green_heart: update chromatic.yml

strawji02 committed:
docs: :memo: update InputBox storybook

InputBox 컴포넌트의 스토리 네이밍을 수정했습니다

strawji02 reviewed (changes requested):
test

strawji02 committed:
Merge pull request #16 from FRIED-NOTE/10-feature-add-storybook

10 feature add storybook

strawji02 committed:
Merge branch '2-feature-add-top-navigation' of https://github.com/My-Recipe/Frontend into 2-feature-add-top-navigation
